generators/typescript/codegen/src/ast/Reference.ts

```diff
@@ -0,0 +1,56 @@
+import { assertNever } from "@fern-api/core-utils";
+import { AstNode } from "./AstNode";
+import { Writer } from "./Writer";
+
+export declare namespace Reference {
+    type Args = NamedImport | ModuleImport;
+
+    interface NamedImport {
+        type: "named";
+        /* The package or path to import from  */
+        source: string;
+        /* Name of the reference to import */
+        name: string;
+    }
+
+    interface ModuleImport {
+        type: "module";
+        /* The package or path to import from  */
+        source: string;
+        /* The module to import from */
+        module: string;
+        /**
+         * The path to access the reference from the module.
+         * [module, ...name].join(".") is the reference
+         */
+        name: string[];
```

Comment:
Should this be named `path`? And should we separate the final element into its own property named `name`? So the reference would effectively be:

```
[module, ...path, name].join(".")
```

If this is too much of a pain, happy to leave this as-is named `name`.

Reply:
this makes sense, will make the change